includes/class-wc-subscriptions-order.php

```diff
@@ -421,16 +425,24 @@ public static function add_contains_subscription_column( $columns ) {
 	public static function add_contains_subscription_column_content( $column ) {
 		global $post;
 
-		if ( 'subscription_relationship' == $column ) {
-			if ( wcs_order_contains_subscription( $post->ID, 'renewal' ) ) {
-				echo '<span class="subscription_renewal_order tips" data-tip="' . esc_attr__( 'Renewal Order', 'woocommerce-subscriptions' ) . '"></span>';
-			} elseif ( wcs_order_contains_subscription( $post->ID, 'resubscribe' ) ) {
-				echo '<span class="subscription_resubscribe_order tips" data-tip="' . esc_attr__( 'Resubscribe Order', 'woocommerce-subscriptions' ) . '"></span>';
-			} elseif ( wcs_order_contains_subscription( $post->ID, 'parent' ) ) {
-				echo '<span class="subscription_parent_order tips" data-tip="' . esc_attr__( 'Parent Order', 'woocommerce-subscriptions' ) . '"></span>';
-			} else {
-				echo '<span class="normal_order">&ndash;</span>';
-			}
+		if ( 'subscription_relationship' === $column ) {
+			self::render_contains_subscription_column_content( $post->ID );
+		}
+	}
+
+	/**
+	 * For HPOS - Add column content to the WooCommerce -> Orders admin screen to indicate whether an
+	 * order is a parent of a subscription, a renewal order for a subscription, or a
+	 * regular order.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string   $column_name Identifier for the custom column.
+	 * @param WC_Order $order       Current WooCommerce order object.
+	 */
+	public static function add_contains_subscription_column_content_hpos( string $column_name, WC_Order $order ) {
```

Comment:
I evaluated using the existing `add_contains_subscription_column_content` method (https://github.com/Automattic/woocommerce-subscriptions-core/pull/497/files#diff-46d20564b0f606d18315a026ac920ffed202942aa778e7a0c2cb21ddfdba67f6R425) for both HPOS and non-HPOS hooks, but ended up adding a separate method to keep the HPOS and non-HPOS paths clearer. Glad to handle it differently.

Reply:
I really like this approach with keeping the logic separate :100:

I've been mulling over whether or not to share this feedback as it seems very nit-picky but I'll share it anyway to get your thoughts.

Since High-Performance Order Storage (HPOS) will eventually become the default option in WC Core, I'm wondering if we should try and avoid introducing functions with "hpos" in the name (unless it's a variable or temporary function that we'll eventually deprecate)

At the moment, adding "hpos" to the name is a nice and short suffix/prefix that gives a lot of context to the function, but as this feature rolls out and becomes the default in WC Core, `_hpos()` may lose a lot of its meaning and this feature might even be changed to something else.

Some alternatives:
- `add_contains_subscription_column_content_list_table()`
- `add_contains_subscription_column_content_orders_table()`
- `add_contains_subscription_custom_column_content()`

With the `_hpos()` removed, these alternatives are very close to the existing non-HPOS function `add_contains_subscription_column_content()` but I think we can remove any confusion by adding some details to the function docblock i.e. "When HPOS is enabled, see @ insert function name"

Reply:
> I've been mulling over whether or not to share this feedback as it seems very nit-picky

Glad you shared your thoughts 🙂 I don't think it was nitpicking, but all potential nit-picks are welcome! 

Makes complete sense. Thanks for noting that!

I updated the method name to `add_contains_subscription_column_content_orders_table` in https://github.com/Automattic/woocommerce-subscriptions-core/pull/497/commits/649dee3dcd972382815ced391546f06ad59a67b3, and updated the docblocks in https://github.com/Automattic/woocommerce-subscriptions-core/pull/497/commits/b966c0b5aa5b762c9fa6a60772d2d579947bb118